bin/propolis-utils/src/bin/cpuid-gen.rs

```diff
@@ -0,0 +1,249 @@
+use std::cmp::{Ord, Ordering};
+use std::collections::BTreeMap;
+
+use bhyve_api::{VmmCtlFd, VmmFd};
+use clap::Parser;
+
+fn create_vm() -> anyhow::Result<VmmFd> {
+    let name = format!("cpuid-gen-{}", std::process::id());
+    let mut req = bhyve_api::vm_create_req::new(&name);
+
+    let ctl = VmmCtlFd::open()?;
+    let _ = unsafe { ctl.ioctl(bhyve_api::VMM_CREATE_VM, &mut req) }?;
+
+    let vm = VmmFd::open(&name).map_err(|e| {
```

Comment:
Could we extend the create request to also return a handle to the newly created vm (say with a new create flag)?

Reply:
As in part of the higher-level API of `bhyve_api` or the underlying ioctl itself?  I have considered adding it to the VM creation ioctl, but that'll probably wait until it can be part of [#15124](https://www.illumos.org/issues/15124)

Reply:
Filed [#15396](https://www.illumos.org/issues/15396) in regards to this.